frontend/src/reducers/user.js

```diff
@@ -0,0 +1,140 @@
+  },
+}
+
+export const user = createSlice({
+  name: 'user',
+  initialState: initialState,
+  reducers: {
+    setAccessToken: (state, action) => {
+      const { accessToken } = action.payload
+      state.login.accessToken = accessToken
+      localStorage.setItem('accessToken', accessToken)
+    },
+    setUserName: (state, action) => {
+      const { userName } = action.payload
+      state.login.userName = userName
+      localStorage.setItem('userName', userName)
+    },
+    setSecretMessage: (state, action) => {
+      const { secretMessage } = action.payload
+      state.login.secretMessage = secretMessage
+    },
+    setErrorMessage: (state, action) => {
+      const { errorMessage } = action.payload
+      state.login.errorMessage = errorMessage
+    },
+  },
+})
+
+//____________THUNKS____________//
+//_______Sign up
+export const signUp = (name, email, password) => {
+  const SIGNUP_URL = "https://the-secret.herokuapp.com/signup"
+  return (dispatch) => {
+    fetch(SIGNUP_URL, {
+      method: 'POST',
+      body: JSON.stringify({ name, email, password }),
+      headers: { 'Content-Type': 'application/json' },
+    })
+      .then((res) => {
+        //Failed
+        if (!res.ok) {
+          throw new Error('Could not create account')
+        }
+        //Success
+        return res.json()
+      })
+      .then((json) => {
+        // Save the login info
+        dispatch(
+          user.actions.setAccessToken({
+            accessToken: json.accessToken,
+          })
+        )
+        dispatch(user.actions.setUserName({ userName: json.userName }))
+      })
+      .catch((err) => {
+        dispatch(user.actions.setErrorMessage({ errorMessage: err.toString() }))
+      })
+  }
+}
+
+//_______Login
+export const login = (name, email, password) => {
+  const LOGIN_URL = 'https://the-secret.herokuapp.com/login'
+  return (dispatch) => {
+    fetch(LOGIN_URL, {
+      method: 'POST',
+      body: JSON.stringify({ name, email, password }),
+      headers: { 'Content-Type': 'application/json' },
+    })
+      .then((res) => {
+        //Success
+        if (res.ok) {
+          return res.json()
+        }
+        // Not OK - failed
+        throw new Error('Unable to sign in. Please check your username and password are correct')
+      })
+      .then((json) => {
+        // Save the login info
+        dispatch(
+          user.actions.setAccessToken({
+            accessToken: json.accessToken,
+          })
+        )
+        dispatch(user.actions.setUserName({ userName: json.userName }))
+      })
+      .catch((err) => {
+        dispatch(logout())
+        dispatch(user.actions.setErrorMessage({ errorMessage: err.toString() }))
+      })
+  }
+}
+
+//_______Secret message
+export const getSecretMessage = () => {
+  const USERS_URL = 'https://the-secret.herokuapp.com/users'
+  return (dispatch, getState) => {
+    const accessToken = getState().user.login.accessToken
+    const userId = getState().user.login.userId
```

Comment:
You are using id here, but after refreshing the page it is no longer in the state 😢